Marketplace/src/main/java/com/example/demo/category/ProductQuery.java

```diff
@@ -0,0 +1,38 @@
+package com.example.demo.category;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class ProductQuery {
+    private long categoryId;
+    private List<Long> characteristics;
+    private OrderBy orderBy;
+    long startId;
+    long endId;
+
+    public ProductQuery(long categoryId, List<Long> characteristics, OrderBy orderBy, long startId, long endId) {
+        this.categoryId = categoryId;
+        this.characteristics = characteristics;
+        this.orderBy = orderBy;
+        this.startId = startId;
+        this.endId = endId;
+    }
+
+    public OrderBy getOrderBy() {
+        return orderBy;
+    }
+
+    public String getCharacteristicsInsertParametersTemplate() {
+        return String.join(",", Collections.nCopies(characteristics.size(), "?"));
```

Comment:
Make convertation to sql in repository layer. You can create separate converter from query to sql